kytos/core/user.py

```diff
@@ -29,13 +29,14 @@ class UserDoc(DocumentBaseModel):
     """UserDocumentModel."""
 
     username: constr(min_length=1, max_length=64, regex="^[a-zA-Z0-9_-]+$")
+    salt: bytes
```

Comment:
@Alopalao the implementation looks good, but let's also structure it in a way that if the rest of `scrypt`'s parameters need to be changed then we don't break compatibility. We probably won't make this configurable in the short term, but later on if we expose this as a configuration to either slightly increase or decrease the complexity of the parameters then we don't break the data model and existing user hash comparisons. With that said, let's also store the rest of parameters (`n`, `r`, `p`) in additoin to the `salt`, feel free to introduce a sub/nested dict in the `UserDoc`

Reply:
Introduced.

Reply:
Looks good. Thanks @Alopalao. Could you also rename `HashDoc` as `HashSubDoc`? Our team has been using this convention of naming `SubDoc` prefix for models that aren't a document but are sub/nested documents on MongoDB. Here's an example on [`flow_manager`](https://github.com/kytos-ng/flow_manager/blob/master/db/models/__init__.py#L91)

Reply:
Oh, I haven't noticed. Changed.